src/dafny/DafnyCore.md

```diff
@@ -0,0 +1,51 @@
+
+# Guide to the Dafny Core Library
+
+The Dafny programming language and tool set includes
+libraries that provide functionality in various domains. This document describes 
+the _Dafny Core Library_. Dafny programs are composed of modules; to ensure
+verification soundness, Dafny imposes a strict non-circular dependency relationship
+among modules. This Core library contains functionality that may be needed
+by other libraries, but it itself is standalone.
+
+All the supported, delivered libraries will
+- be stable --- a minimum of backwards incompatible changes
+- be verified --- proved to meet specifications and tested with expected use cases,
+with each release of the library and of the Dafny toolset
+- be efficient --- efficient in both proofs and executables that use the libraries
+- be compilable for every supported target language by default, with exceptions being documented and for exceptional reasons
```

Comment:
I love these goals but don't want to state them as achieved like this since we don't have mechanisms to back them up yet (especially the latter two), so I'll probably push back on this section even in #92.

Reply:
I'll state them as the goals we are aiming for -- which is what I intended.